charts/telegraf/templates/service.yaml

```
@@ -62,6 +62,12 @@ spec:
     targetPort: {{ trimPrefix ":" $value.service_address | int64 }}
     name: "webhooks"
     {{- end }}
+    {{- if eq $key "syslog" }}
+  - port: {{ regexFind "[0-9]+$" $value.server  | int64 }}
```

To make it safe I'd suggest to add protocol condition check (with unix socket server address the rendered port spec is invalid), eg.

```suggestion
    {{- if eq $key "syslog" }}
      {{- if regexMatch "^(tcp|udp).*" $value.server }}
  - port: {{ regexFind "[0-9]+$" $value.server  | int64 }}
```